cmd/vcert/validators.go

```diff
@@ -75,6 +76,9 @@ func readData(commandName string) error {
 
 func validateCommonFlags(commandName string) error {
 
+	if !httpguts.ValidHeaderFieldValue(flags.userAgent) {
+		return fmt.Errorf("Invalid user-agent value: %q", flags.userAgent)
+	}
```

Comment:
This is the same validation that is performed by the Go http.Client before it writes the headers to the network:
 * https://github.com/golang/go/blob/5f5b20c4268c1a3aa6a3b132aeede6dc82adf344/src/net/http/transport.go#L516-L530